app/scripts/modules/core/src/pipeline/executions/execution/ExecutionMarkerInformationPopover.tsx

```diff
@@ -0,0 +1,360 @@
+  private showPipelineGraph = () => {
+    this.setState({ showPipelineGraph: true });
+  };
+
+  private hidePipelineGraph = () => {
+    this.setState({ showPipelineGraph: false });
+  };
+
+  private toggleParameters = (target: any) => {
+    const container = target.nextElementSibling;
+    const icon = target.children[0];
+    // since we are toggle to the next stage reverse condiontal here
+    const isClosed = container.classList.contains('closed');
+
+    const iconClass = classNames({
+      'fa-chevron-down': isClosed,
+      'fa-chevron-right': !isClosed,
+      fa: true,
+    });
+    const containerClass = classNames({
+      'view-parameters': true,
+      closed: !isClosed,
+      opened: isClosed,
+    });
+
+    icon.className = iconClass;
+    container.className = containerClass;
+  };
+
+  public render(): React.ReactElement<HTMLDivElement> {
+    const { executionDetails, failedInApplication, link, pipelineConfigDetails, stageDetails } = this.state;
+
+    const content = this.state.showPipelineGraph ? (
+      <div className="">
+        {this.allExecutions.map(item => {
+          return (
+            <div key={`${item.execution.id}-${item.execution.name}`} className="execution-graph">
+              {item.execution.application} - {item.execution.name}
+              <PipelineGraph
+                key={item.execution.id}
+                execution={item.execution}
+                onNodeClick={() => {}}
+                viewState={{
+                  activeStageId: item.stageIndex,
+                  activeSubStageId: null,
+                  canTriggerPipelineManually: false,
+                  canConfigure: false,
+                }}
+              />
+            </div>
+          );
+        })}
+      </div>
+    ) : (
+      <div>
+        {!executionDetails ? (
+          <div>
+            <Spinner size="medium" />
+          </div>
+        ) : (
+          <div className="information-details">
+            <div className="pipeline-name bottom-margin">
+              {pipelineConfigDetails ? pipelineConfigDetails.name : '-'}
+            </div>
+            <div style={{ fontWeight: 'bold' }}>PIPELINE</div>
+            <div className="bottom-margin">
+              <div>{`[${executionDetails.authentication.user}] (${executionDetails.user})`}</div>
+              <div>{relativeTime(executionDetails.startTime || executionDetails.buildTime)}</div>
+              <div>
+                Status: <span className="status">{executionDetails.status}</span> by parent pipeline{' '}
+                {timestamp(executionDetails.trigger.parentExecution.buildTime)}
+              </div>
+            </div>
+            <div className="bottom-margin parameters">
+              <span
+                className="clickable"
+                onClick={event => {
+                  this.toggleParameters(event.target);
+                }}
+              >
+                <i className="fa fa-chevron-right"></i> View all Parameters (
```

Comment:
Can we use the existing [`ParametersAndArtifacts`](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/pipeline/status/ParametersAndArtifacts.tsx) component that is used in the main execution details instead of building our own here? That should handle the chevron manipulation for you, so we can remove the `toggleParameters` method, which I'm not sure is currently working as expected since I'm currently seeing the following error when clicking there:

```
Uncaught TypeError: Cannot read property 'classList' of null```

Reply:
i'll check out that component thanks